### PR TITLE
HOTFIX SOLVING NONE DoF ERROR: Moving corrected version node.h to master branch

### PR DESCRIPTION
**Pull request by loumalouomega**
I can not find the original commit to cherrypick, I upload directly the final version

### CODE REVIEW

**Comment by loumalouomega** [comment]
Concerning #495

**Comment by josep-m-carbonell** [comment]
@loumalouomega  no files changed ...

This method has to be corrected too: @pooyan-dadvand  (see the NOTE)

    /** adds a Dof to the node and return new added dof or existed one. */
    inline typename DofType::Pointer pAddDof(DofType const& SourceDof)
    {
        KRATOS_TRY_LEVEL_3

        typename DofsContainerType::iterator i_dof = mDofs.find(SourceDof.GetVariable());

	//NOTE: here returns the same Dof if exists, but if the SourceDof has a reaction and the
	//found dof does not has it, then the reaction is not set!!!
        if(i_dof != mDofs.end())
            return *(i_dof.base());

        typename DofType::Pointer p_new_dof =  boost::make_shared<DofType>(SourceDof);
        mDofs.insert(mDofs.begin(), p_new_dof);

        p_new_dof->SetId(Id());

        p_new_dof->SetSolutionStepsData(&mSolutionStepsNodalData);

        mDofs.Sort();

        return p_new_dof;

        KRATOS_CATCH_LEVEL_3(*this);

    }

**Comment by loumalouomega** [comment]
@josep-m-carbonell This is probably an error, I uploaded from the phone and the information was partial, I will check it now.

**Comment by loumalouomega** [comment]
@josep-m-carbonell Indeed, the file changed was the node.h, not the model_part.h. Uploading. Sorry for the inconvenience

File: kratos/includes/node.h [hunk 1]
```diff
@@ -789,7 +792,7 @@ class Node : public Point<TDimension>,  public IndexedObject, public Flags
     	{
     	  typename SolutionStepsNeighboursContainerType::iterator i;
     	  if((i = mSolutionStepsNeighbours.find(SolutionStepIndex)) == mSolutionStepsNeighbours.end())
-    	    KRATOS_THROW_ERROR(std::invalid_argument, "Solution step index out of range.", *this);
+    	    KRATOS_ERROR <<  "Solution step index out of range.", *this);
```
Comment (RiccardoRossi) [inline kratos/includes/node.h]: this can not compile...

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
what is the reason of those changes? 
i would avoid cosmetic changes in the base class in the even they introduce errors. (in this specific case i think there is one with a KRATOS_ERROR which can not compile.

**Comment by loumalouomega** [comment]
@RiccardoRossi The KRATOS_ERROR you referred it was in a commented piece of code, so it didn't affected the compilation.

**Comment by RiccardoRossi** [comment]
@josep-m-carbonell 

what you say is correct, the code should look something like

        if(i_dof != mDofs.end())
        {
             if(i_dof->GetReaction() != SourceDof.GetReaction())
                  i_dof->SetReaction(SourceDof.GetReaction());
            return *(i_dof.base());
       }

however such code does not compile (better said, it does not link).

@pooyan-dadvand what would be the correct way of doing this?

File: kratos/includes/node.h [hunk 2]
```diff
@@ -1043,8 +1043,16 @@ class Node : public Point<TDimension>,  public IndexedObject, public Flags
 
 
         typename DofsContainerType::iterator itDoF = mDofs.find(SourceDof.GetVariable());
-        if(itDoF != mDofs.end())
-            return *(itDoF.base());
+        if(i_dof != mDofs.end())
```
Comment (loumalouomega) [inline kratos/includes/node.h]: @RiccardoRossi It is called itDoF not i_dof

**Review by loumalouomega — COMMENTED** [review]
I corrected the i_dof with itDoF

**Comment by RiccardoRossi** [comment]
@loumalouomega, thank you

**Review by RiccardoRossi — APPROVED** [review]

File: kratos/includes/node.h [hunk 3]
```diff
@@ -329,70 +328,72 @@ class Node : public Point<TDimension>,  public IndexedObject, public Flags
...
-        omp_init_lock(&mnode_lock);
+        omp_init_lock(&mNodeLock);
 #endif
     }
 
     typename Node<TDimension>::Pointer Clone()
     {
-        Node<3>::Pointer pnew_node = boost::make_shared<Node<3> >( this->Id(), (*this)[0], (*this)[1], (*this)[2]);
-        pnew_node->mSolutionStepsNodalData = this->mSolutionStepsNodalData;
+        Node<3>::Pointer pNewNode = boost::make_shared<Node<3> >( this->Id(), (*this)[0], (*this)[1], (*this)[2]);
```
Comment (pooyan-dadvand) [inline kratos/includes/node.h:342]: local variables are lower case with underline: `p_new_node`

Comment (loumalouomega) [reply kratos/includes/node.h:342]: OK

Comment (loumalouomega) [reply kratos/includes/node.h:342]: Do I reopen this?

Comment (RiccardoRossi) [reply kratos/includes/node.h:342]: i would not touch this again...

Comment (loumalouomega) [reply kratos/includes/node.h:342]: I am trying to do things right...

**Pull request merged**